services/headless-lms/chatbot/src/chatbot_tools/course_progress.rs

```diff
@@ -0,0 +1,320 @@
+                score_required: Some(5),
+                score_maximum: Some(6),
+                total_exercises: Some(6),
+                attempted_exercises: Some(4),
+                attempted_exercises_required: Some(5),
+            },
+            UserCourseProgress {
+                course_module_id: Uuid::nil(),
+                course_module_name: "Chatbot advanced topics".to_string(),
+                course_module_order_number: 4,
+                score_given: 2.0,
+                score_required: None,
+                score_maximum: None,
+                total_exercises: Some(2),
+                attempted_exercises: Some(2),
+                attempted_exercises_required: None,
+            },
+        ];
+
+        let tool = CourseProgressTool::new_mock("Advanced Chatbot Course".to_string(), progress);
+        let output = tool.get_tool_output();
+
+        let expected_output = "Result: [output]The user is completing a course called Advanced Chatbot Course. The user's progress on the base course module called Advanced Chatbot Course is the following: On this module, there are a total of 5 exercises and 10 exercise points. To pass this module, it's required to attempt 5 exercises and gain 8 exercise points. The user has attempted 5 exercises. They have attempted enough exercises to pass this module if they have also received enough points. The user has gained 8.0 points. The user has gained enough points to pass this module. \nTo pass the course, it's required to pass the base module. The following modules are additional to the course and to complete them, it's required to first complete the base module. \nThe user's progress on the course module called First extra module is the following: On this module, there are a total of 6 exercises and 6 exercise points. To pass this module, it's required to attempt 5 exercises and gain 5 exercise points. The user has attempted 4 exercises. To pass this module, they need to attempt 1 more exercises. The user has gained 3.9 points. To pass this module, the user needs to gain 1.1 more points. \nThe user's progress on the course module called Second extra module is the following: On this module, there are a total of 6 exercises and 5 exercise points. To pass this module, it's required to attempt 5 exercises and gain 4 exercise points. The user has not attempted any exercises. To pass this module, they need to attempt 5 more exercises. The user has gained 0.0 points. To pass this module, the user needs to gain 4.0 more points. \nThe user's progress on the course module called Chatbot advanced topics is the following: On this module, there are a total of 2 exercises. The user has attempted 2 exercises. The user has gained 2.0 points. \n[/output] \n\nInstructions for describing the output: [instructions]Describe this information in a short, clear way with no or minimal bullent points. Only give information that is relevant to the user's question.[/instructions]".to_string();
```

Comment:
Remove this: "To pass the course, it's required to pass the base module.".

Then add after the first sentence: "The course has one base module, and additional modules. The base module can be completd on its own, and the additional modules require the base module to be completed before they can be completed."